### PR TITLE
[aot] [vulkan] Add AotKernel and its Vulkan impl

### PR DESCRIPTION
**Pull request by k-ye**
Related issue = https://github.com/taichi-dev/taichi/issues/3642

### CODE REVIEW

File: taichi/backends/vulkan/aot_module_loader_impl.cpp [hunk 1]
```diff
@@ -3,9 +3,30 @@
...
+      : runtime_(runtime), handle_(handle) {
+  }
+
+  void run(RuntimeContext *ctx) override {
+    runtime_->launch_kernel(handle_, ctx);
+  }
+
+ private:
+  VkRuntime *const runtime_;
+  const KernelHandle handle_;
```
Comment (bobcao3) [inline taichi/backends/vulkan/aot_module_loader_impl.cpp:27]: Out of curiosity, I wonder maybe we can ditch the KernelHandle thing and use some pure data structure to replace it?

**Review by bobcao3 — APPROVED** [review]
LGTM!

File: taichi/backends/vulkan/aot_module_loader_impl.cpp (hunk 1 above)
Comment (k-ye) [reply taichi/backends/vulkan/aot_module_loader_impl.cpp:27]: I do prefer `KernelHandle` for type safety. It's identical to an integer, so no runtime overhead :-)

**Pull request merged**